FastBank.Infrastructure/Repository/UserRepository.cs

```diff
@@ -42,7 +43,32 @@ public void Add(User user)
                     user.Role,
                     user.Inactive);
 
-            _repo.Add(userDTO);           
+            _repo.Add(userDTO);
+        }
+
+        public List<User> GetUserFriends(User user)
+        {
+            var friends = _repo.SetNoTracking<FriendsRelationDTO>()
+                                    .Where(u => u.UserId == user.Id)
+                                    .Include(u => u.User)
+                                    .Include(u => u.Friend)
+                                    .Select(u => u.Friend.ToDomainObj())
+                                    .ToList();
+            return friends;
+        }
+
+        public void AddFriend(User user, User friend)
+        {
+            _repo.Add<FriendsRelationDTO>(new FriendsRelationDTO(Guid.NewGuid(), user, friend, false));
```

Comment:
Do you have a logic for checking if there is already friendship, let's say
Angel - Alex - stored in DB
and we try to add new friendship
Angel - Alex ?
We should prevent any dublication in our database, no matter of what type it is

Reply:
Yes I have this logic.
![image](https://github.com/bulgarchik/FastBank/assets/41870483/7e31e772-454c-4ace-9ece-7a13b3deda71)